Fix: Include RLS filters for cache keys

gtg472b opened the pull request:
This fix makes sure that RLS filters are searched for templatable jinja content, ensuring cached visualizations aren't shown to the wrong user.

### SUMMARY
I added a check to see if there are any RLS filters that are templated

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TEST PLAN
I haven't tried it with multiple layers of RLS (eg, a user has multiple roles, and each role has different RLS filters), so that may need to be tested.

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [x] Has associated issue: Fixes #10804 
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

gtg472b committed:
Fix: Include RLS filters for cache keys

This fix makes sure that RLS filters are searched for templatable jinja content, ensuring cached visualizations aren't shown to the wrong user.

villebro reviewed (approved):
Good catch @gtg472b 👍

villebro commented on superset/connectors/sqla/models.py:
```diff
@@ -1454,6 +1454,11 @@ def has_extra_cache_key_calls(self, query_obj: QueryObjectDict) -> bool:
             templatable_statements.append(extras["where"])
         if "having" in extras:
             templatable_statements.append(extras["having"])
+        # make sure the RLS filters are included
+        if config["ENABLE_ROW_LEVEL_SECURITY"] and self.is_rls_supported:
+            templatable_statements += [
+                f.clause
+                for f in security_manager.get_rls_filters(self) ]
```
Please make sure to run black on your code prior to opening a PR (check `pre-commit` hooks in [CONTIBUTING.md](https://github.com/apache/incubator-superset/blob/master/CONTRIBUTING.md#git-hooks))
```suggestion
        if config["ENABLE_ROW_LEVEL_SECURITY"] and self.is_rls_supported:
            templatable_statements += [
                f.clause for f in security_manager.get_rls_filters(self)
            ]
```

gtg472b committed:
Fix: Include RLS filters for cache keys

This fix makes sure that RLS filters are searched for templatable jinja content, ensuring cached visualizations aren't shown to the wrong user.

Co-authored-by: Ville Brofeldt <33317356+villebro@users.noreply.github.com>

gtg472b commented:
Thanks for fixing it @villebro . I don't have a dev system set up for Superset, and I almost never do anything in Python. Never use GitHub either (this is my first PR!)

villebro reviewed (approved):
In that case you've done a stellar job @gtg472b! LGTM, going to merge as soon as CI finishes 🎉
